downstairs/src/lib.rs

```diff
@@ -822,10 +860,10 @@ where
                                 upstairs_connection, ds.jobs().await,
                             );
 
-                            if ds.is_active(upstairs_connection.upstairs_id) {
+                            if ds.is_active(upstairs_connection) {
```

Comment:
We have the ds.lock at this point, right?  So we would have to be active to call `ds.clear.active()` here?

Reply:
Yes, you're right - there's no bug here. This Upstairs would have to be active.